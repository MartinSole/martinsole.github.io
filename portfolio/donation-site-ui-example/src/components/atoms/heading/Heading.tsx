interface Props {
    level: number
    children: string
}

const Heading = ({level, children}: Props) => {
    const CustomTag = `h${level}` as keyof JSX.IntrinsicElements;

  return (
    <CustomTag>{children}</CustomTag>
  )
}

export default Heading