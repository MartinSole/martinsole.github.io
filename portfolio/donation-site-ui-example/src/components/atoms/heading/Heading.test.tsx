import '@testing-library/jest-dom'
import { render } from "@testing-library/react"
import Heading from "./Heading"

test("Renders the Heading component", () => {
   const {container, findByText} = render(<Heading children="Heading" level={1} />)
    expect(container).toMatchSnapshot()
    expect(findByText('Heading')).toBeTruthy
})
