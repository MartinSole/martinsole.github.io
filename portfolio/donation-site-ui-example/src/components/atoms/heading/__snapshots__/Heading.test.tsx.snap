// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the Heading component 1`] = `
<div>
  <h1>
    Heading
  </h1>
</div>
`;
